homeassistant/components/foscam/__init__.py

```diff
@@ -81,3 +99,45 @@ def update_unique_id(entry):
     LOGGER.info("Migration to version %s successful", entry.version)
 
     return True
+
+
+class FoscamCoordinator(DataUpdateCoordinator):
+    """My custom coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        session: FoscamCamera,
+        update_interval=timedelta(seconds=30),
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            # Name of the data. For logging purposes.
+            name=DOMAIN,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=update_interval,
+        )
+        self._session = session
+
+    def get_session(self) -> FoscamCamera:
+        """Return the foscam camera session."""
+        return self._session
+
+    async def _async_update_data(self) -> dict:
+        """Fetch data from API endpoint."""
+
+        async with async_timeout.timeout(30):
+            data = {}
+            ret, dev_info = await self.hass.async_add_executor_job(
+                self._session.get_dev_info
```

Comment:
Can these calls raise so we need to add a try-block?

Reply:
No, if it fails an error code will be returned (ret)